### PR TITLE
Add cypress tests for project access page

### PR DESCRIPTION
**Pull request by sjaanus**
Add cypress tests for project access page

### CODE REVIEW

File: cypress/integration/projects/access/project-access.spec.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    PA_EDIT_BUTTON_ID,
+    PA_REMOVE_BUTTON_ID,
+    PA_ROLE_ID,
+    PA_USERS_GROUPS_ID,
+    PA_USERS_GROUPS_TITLE_ID,
+} from '../../../../src/utils/testIds';
+
+export {};
+const baseUrl = Cypress.config().baseUrl;
+const randomId = String(Math.random()).split('.')[1];
```
[the remark this answers is not in the document]
Comment (sjaanus) [reply cypress/integration/projects/access/project-access.spec.ts:15]: @sonatype-lift ignore

File: cypress/integration/segments/segments.spec.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ const disableActiveSplashScreens = () => {
     cy.visit(`/splash/operators`);
 };
 
-describe('segments', () => {
+describe('project-access', () => {
```
Comment (nunogois) [inline cypress/integration/segments/segments.spec.ts:12]: I think this was a mistake that should be reverted 🙂

File: src/component/project/ProjectAccess/ProjectAccessTable/ProjectAccessTable.tsx [hunk 3]
```diff
@@ -43,6 +43,11 @@ import { ProjectAccessCreate } from 'component/project/ProjectAccess/ProjectAcce
 import { ProjectAccessEditUser } from 'component/project/ProjectAccess/ProjectAccessEditUser/ProjectAccessEditUser';
 import { ProjectAccessEditGroup } from 'component/project/ProjectAccess/ProjectAccessEditGroup/ProjectAccessEditGroup';
 import useHiddenColumns from 'hooks/useHiddenColumns';
+import {
+    PA_ASSIGN_BUTTON_ID,
+    PA_EDIT_BUTTON_ID,
+    PA_REMOVE_BUTTON_ID,
+} from '../../../../utils/testIds';
```
Comment (nunogois) [inline src/component/project/ProjectAccess/ProjectAccessTable/ProjectAccessTable.tsx:50]: Should avoid relative paths.
```suggestion
} from 'utils/testIds';
```

File: cypress/integration/projects/access/project-access.spec.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                sendEmail: false,
+                rootRole: 3,
+            })
+                .as(name)
+                .then(response => {
+                    const id = response.body.id;
+                    userIds.push(id);
+                    cy.request('POST', `${baseUrl}/api/admin/groups`, {
+                        name: `${i}-${groupAndProjectName}`,
+                        users: [{ user: { id: id }, role: 'Owner' }],
```
Comment (nunogois) [inline cypress/integration/projects/access/project-access.spec.ts:44]: Should be adapted once https://github.com/Unleash/unleash/pull/1905 gets merged.
```suggestion
                        users: [{ user: { id: id }],
```

**Review by nunogois — CHANGES_REQUESTED** [review]
Good job with this 👍 Just needs attention to some small details.
 - You should also add this as an action to `.github/workflows/e2e.projects.access.yml`, for example, based on existing ones;
 - Might need some slight adjustments due to changes on https://github.com/Unleash/unleash-frontend/pull/1210 as well, depending on which gets merged first;

File: src/component/project/ProjectAccess/ProjectAccessTable/ProjectAccessTable.tsx (hunk 3 above)
Comment (sjaanus) [reply src/component/project/ProjectAccess/ProjectAccessTable/ProjectAccessTable.tsx:50]: Damn the auto import.

File: cypress/integration/segments/segments.spec.ts (hunk 2 above)
Comment (sjaanus) [reply cypress/integration/segments/segments.spec.ts:12]: Thanks. Typo.

File: cypress/integration/projects/access/project-access.spec.ts (hunk 4 above)
Comment (sjaanus) [reply cypress/integration/projects/access/project-access.spec.ts]: Removed

**Review by nunogois — APPROVED** [review]
LGTM, well done!

image

**Pull request merged**